spec/controllers/training_modules_users_controller_spec.rb

```diff
@@ -30,6 +33,17 @@
           .to be_between(1.minute.ago, 1.minute.from_now)
       end
     end
-  end
+    context 'nonexistent slide' do
+      let(:nonexistent_slide_params) do
+        { module_id: t_mod.slug,
+          slide_id: 'not-a-real-slide',
+          user_id: user.id }
+      end
 
+      it 'renders a response' do
+        post :create_or_update, nonexistent_slide_params, format: :json
+        expect(response).to be
```

Comment:
In this case, it's a json request fired by the client to say 'I made it through this module all the way to this slide)', rather than the html request that redirects to a 404. So, at least as I patched it here, it's a 200 rather than a 302. Would it be better to redirect that json request to the 404 page too?